src/python_testing/TC_CGEN_2_2.py

```diff
@@ -0,0 +1,796 @@
+        logger.info('Step  # 33 - Failsafe timer expiration bypassed for TH2 by setting expiryLengthSeconds to 0. Test continues without the original wait.')
+
+        self.step(34)
+        trusted_root_list_original_updated = await self.read_single_attribute_check_success(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cluster=cluster_opcreds,
+            attribute=cluster_opcreds.Attributes.TrustedRootCertificates)
+        trusted_root_list_original_size_updated = len(trusted_root_list_original_updated)
+        logger.info(f'Step #34 - The updated num_trusted_roots_original: {trusted_root_list_original_size_updated}')
+        # Verify that the trusted root list size
+        asserts.assert_equal(trusted_root_list_original_size_updated, trusted_root_list_original_size + 2,
+                             "Step #34 - Unexpected number of entries in the TrustedRootCertificates table after update")
+
+        self.step(35)
+        cmd = Clusters.GeneralCommissioning.Commands.ArmFailSafe(expiryLengthSeconds=maxFailsafe, breadcrumb=1)
+        resp = await self.send_single_cmd(
+            dev_ctrl=TH2,
+            node_id=newNodeId+1,
+            cmd=cmd)
+        # Verify that the DUT responds with ArmFailSafeResponse with ErrorCode as 'BusyWithOtherAdmin'(4)
+        logger.info(f'Step #35 - TH2 ArmFailSafeResponse with ErrorCode as BusyWithOtherAdmin ({resp.errorCode})')
+        asserts.assert_equal(resp.errorCode, Clusters.GeneralCommissioning.Enums.CommissioningErrorEnum.kBusyWithOtherAdmin,
+                             "Failure status returned from arm failsafe")
+
+        self.step(36)
+        cmd = Clusters.GeneralCommissioning.Commands.ArmFailSafe(expiryLengthSeconds=0)
+        resp = await self.send_single_cmd(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cmd=cmd)
+        logger.info(f'Step #36 - response attributes are: {resp.errorCode}')
+        # Verify that the DUT responds with ArmFailSafeResponse with ErrorCode as 'OK'(0)
+        asserts.assert_equal(resp.errorCode, Clusters.GeneralCommissioning.Enums.CommissioningErrorEnum.kOk,
+                             "Failure status returned from arm failsafe")
+
+        self.step(37)
+        cmd = Clusters.GeneralCommissioning.Commands.ArmFailSafe(expiryLengthSeconds=maxFailsafe)
+        resp = await self.send_single_cmd(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cmd=cmd)
+        logger.info(f'Step #37 - response attributes are: {resp.errorCode}')
+        # Verify that the DUT responds with ArmFailSafeResponse with ErrorCode as 'OK'(0)
+        asserts.assert_equal(resp.errorCode, Clusters.GeneralCommissioning.Enums.CommissioningErrorEnum.kOk,
+                             "Failure status returned from arm failsafe")
+
+        self.step(38)
+        # Step 38 - Skipped
```

Comment:
keep this in for the cert testing - it is just on the CI that we're going to avoid waiting.